streams/src/test/java/org/apache/kafka/streams/TopologyTest.java

```diff
@@ -174,6 +174,24 @@ public void shouldNotAllowToAddProcessorWithSameName() {
         } catch (final TopologyException expected) { }
     }
 
+    @Test
+    public void shouldNotAllowToAddProcessorWithEmptyParents() {
+        topology.addSource("source", "topic-1");
+        try {
+            topology.addProcessor("processor", new MockProcessorSupplier());
+            fail("Should throw TopologyException for processor without at least one parent node");
+        } catch (final TopologyException expected) { }
+    }
+
+    @Test
+    public void shouldNotAllowToAddProcessorWithNullParents() {
+        topology.addSource("source", "topic-1");
+        try {
+            topology.addProcessor("processor", new MockProcessorSupplier(), null);
```

Comment:
This does not test what you want to test, but will pass in an array with one element that is `null` -- You need to call as follows:
```
String[] nullParents = null;
topology.addProcessor("processor", new MockProcessorSupplier(), nullParents);
```

the test still throws, because `null` as parent name is not allowed either -- maybe it worth to add a check for the exception error message, too. It would be different for both cases.

Reply:
Also, the test expects a `TopologyException`; if an error is thrown because `null` as a parent name is not allowed it would be a `NullPointerError` from `Objects.requireNonNull`. This would cause the test to fail if passing in `null` results in `new String[] {null}`.

Reply:
Interesting. Thanks for clarifying. Makes sense. Seem my understanding how varargs work was wrong.

With regard to testing for `null`. For the other parameters, we actually use `Objects.requireNonNull(...)` -- thus, I am wondering if it might be better (for consistency reasons) to do the same for the `parentName` parameter?

Reply:
Yeah, that makes sense! I chose to put it in the if statement because it was an easy way to repeat the string, but actually now I think about it a different string would be fine since passing null is different from passing no args. I'll make the change.